src/suites/cts/validation/createBindGroup.spec.ts

```diff
@@ -2,94 +2,36 @@ export const description = `
 createBindGroup validation tests.
 `;
 
-import { TestGroup, pcombine, poptions, unreachable } from '../../../framework/index.js';
+import { C, TestGroup, pbool, pcombine, poptions, unreachable } from '../../../framework/index.js';
+import { bindingTypes } from '../format_info.js';
 
-import { ValidationTest } from './validation_test.js';
+import { BindingResourceType, ValidationTest } from './validation_test.js';
 
 function clone(descriptor: GPUTextureDescriptor): GPUTextureDescriptor {
   return JSON.parse(JSON.stringify(descriptor));
 }
 
-class F extends ValidationTest {
-  getStorageBuffer(): GPUBuffer {
-    return this.device.createBuffer({
-      size: 1024,
-      usage: GPUBufferUsage.STORAGE,
-    });
-  }
-
-  getUniformBuffer(): GPUBuffer {
-    return this.device.createBuffer({
-      size: 1024,
-      usage: GPUBufferUsage.UNIFORM,
-    });
-  }
-
-  getSampler(): GPUSampler {
-    return this.device.createSampler();
-  }
-
-  getSampledTexture(): GPUTexture {
-    return this.device.createTexture({
-      size: { width: 16, height: 16, depth: 1 },
-      format: 'rgba8unorm',
-      usage: GPUTextureUsage.SAMPLED,
-    });
-  }
-
-  getStorageTexture(): GPUTexture {
-    return this.device.createTexture({
-      size: { width: 16, height: 16, depth: 1 },
-      format: 'rgba8unorm',
-      usage: GPUTextureUsage.STORAGE,
-    });
-  }
-}
-
-export const g = new TestGroup(F);
+export const g = new TestGroup(ValidationTest);
 
 g.test('binding count mismatch', async t => {
   const bindGroupLayout = t.device.createBindGroupLayout({
-    bindings: [
-      {
-        binding: 0,
-        visibility: GPUShaderStage.COMPUTE,
-        type: 'storage-buffer',
-      },
-    ],
+    bindings: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type: 'storage-buffer' }],
```

Comment:
Can `'storage-buffer'` be `C.BindingType.StorageBuffer` ?

Reply:
Could, but I don't think it's helpful. These keys are already tightly typed by `[k in GPUBindingType]` so typos aren't possible. In fact, that forces us to be complete too (no missing variants). Also the original string values are shorter and I think easier to read.

Reply:
yes, though it doesn't make any difference in this context (inference can determine this is a GPUBindingType so it's already type/typo-checked) and only really serves to make it longer